From 1954c22a1da3416a0d709e2f6827fbaadcf81fed Mon Sep 17 00:00:00 2001
From: Swarnim Kulkarni <kulkarni.swarnim@gmail.com>
Date: Tue, 8 Sep 2015 21:59:00 -0800
Subject: [PATCH 540/587] CDH-31761: CDH-38878: HIVE-11590 : AvroDeserializer
 is very chatty (Swarnim Kulkarni via Xuefu Zhang)

Conflicts:
	serde/src/java/org/apache/hadoop/hive/serde2/avro/AvroSerDe.java

Change-Id: I044a8bce6b216b9dea8deda29c160368187f55a0
---
 .../apache/hadoop/hive/serde2/avro/AvroSerDe.java  |   13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/serde/src/java/org/apache/hadoop/hive/serde2/avro/AvroSerDe.java b/serde/src/java/org/apache/hadoop/hive/serde2/avro/AvroSerDe.java
index efff663..0e4e4c6 100644
--- a/serde/src/java/org/apache/hadoop/hive/serde2/avro/AvroSerDe.java
+++ b/serde/src/java/org/apache/hadoop/hive/serde2/avro/AvroSerDe.java
@@ -84,7 +84,7 @@ public void initialize(Configuration configuration, Properties tableProperties,
   public void initialize(Configuration configuration, Properties properties) throws SerDeException {
     // Reset member variables so we don't get in a half-constructed state
     if (schema != null) {
-      LOG.info("Resetting already initialized AvroSerDe");
+      LOG.debug("Resetting already initialized AvroSerDe");
     }
 
     schema = null;
@@ -110,10 +110,12 @@ public void initialize(Configuration configuration, Properties properties) throw
       properties.setProperty(AvroSerdeUtils.AvroTableProperties.SCHEMA_LITERAL.getPropName(), schema.toString());
     }
 
-    LOG.info("Avro schema is " + schema);
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Avro schema is " + schema);
+    }
 
     if (configuration == null) {
-      LOG.info("Configuration null, not inserting schema");
+      LOG.debug("Configuration null, not inserting schema");
     } else {
       configuration.set(
           AvroSerdeUtils.AvroTableProperties.AVRO_SERDE_SCHEMA.getPropName(), schema.toString(false));
@@ -136,7 +138,10 @@ public static Schema getSchemaFromCols(Properties properties,
       //Comments are separated by "\0" in columnCommentProperty, see method getSchema
       //in MetaStoreUtils where this string columns.comments is generated
       columnComments = Arrays.asList(columnCommentProperty.split("\0"));
-      LOG.info("columnComments is " + columnCommentProperty);
+
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("columnComments is " + columnCommentProperty);
+      }
     }
     if (columnNames.size() != columnTypes.size()) {
       throw new IllegalArgumentException("AvroSerde initialization failed. Number of column " +
-- 
1.7.9.5

